src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CollectionExpression.cs

```diff
@@ -738,6 +957,24 @@ private BoundExpression CreateAndPopulateList(BoundCollectionExpression node, Ty
                         // list.Add(element);
                         expressions.Add(
                             _factory.Call(listTemp, addMethod, rewrittenValue));
+                    },
+                    (sideEffects, spanTemp, spreadElement, rewrittenSpreadOperand) =>
+                    {
+                        if (addRangeMethod is null)
+                            return false;
+
+                        var type = rewrittenSpreadOperand.Type!;
+
+                        var useSiteInfo = GetNewCompoundUseSiteInfo();
+                        var conversion = _compilation.Conversions.ClassifyConversionFromType(type, addRangeMethod.Parameters[0].Type, isChecked: false, ref useSiteInfo);
+                        _diagnostics.Add(rewrittenSpreadOperand.Syntax, useSiteInfo);
+                        if (conversion.IsImplicit && conversion.IsReference)
+                        {
+                            sideEffects.Add(_factory.Call(listTemp, addRangeMethod, MakeConversionNode(rewrittenSpreadOperand, type, @checked: false, markAsChecked: true)));
```

Comment:
>MakeConversionNode(rewrittenSpreadOperand, type, @checked: false, markAsChecked: true) [](http://example.com/codeflow?start=84&length=86)

We've already calculated the `conversion` above. Could we call the following instead?
```csharp
conversion.MarkUnderlyingConversionsCheckedRecursive();
sideEffects.Add(_factory.Call(listTemp, addRangeMethod,
    MakeConversionNode(rewrittenSpreadOperand.Syntax, rewrittenSpreadOperand, conversion, type, checked: false)));
```